File: Algorithms/Python/pattern_searching/brute_force.py
from __future__ import annotations


def search_pattern(text: str, pattern: str) -> int:
    for i in range(len(text) - len(pattern) + 1):
        j = 0
        while j < len(pattern) and text[i + j] == pattern[j]:
            j += 1
        if j == len(pattern):
            return i

    return -1


if __name__ == "__main__":
    text = "CAABAACAADAABAAABAA"
    pattern = "AABA"
    print("Pattern found at index", search_pattern(text, pattern))

File: Algorithms/Javascript/pattern_searching/bruteForce.js
const searchPattern = (text, pattern) => {
    const n = text.length;
    const m = pattern.length;
    
    for (let i = 0; i <= n - m; i++) {
        let j;
        for (j = 0; j < m; j++) {
            if (text[i + j] !== pattern[j]) {
                break;
            }
        }
        if (j === m) {
            console.log(`Pattern found at index ${i}`);
        }
    }
}

const text = 'AABAACAADAABAAABAA';
const pattern = 'AABA';

searchPattern(text, pattern);
